clients/drcachesim/tracer/tracer.cpp

```diff
@@ -458,8 +456,13 @@ memtrace(void *drcontext, bool skip_size_cap)
                                                 (pipe_start+header_size)));
                 atomic_pipe_write(drcontext, pipe_start, buf_ptr);
             }
+        } else {
+            write_trace_data(drcontext, pipe_start, buf_ptr);
         }
-        data->num_refs += num_refs;
+        auto span = buf_ptr - (data->buf_base + header_size);
+        DR_ASSERT(0 == span % instru->sizeof_entry());
```

Comment:
style: x==0, not 0==x: see https://github.com/DynamoRIO/dynamorio/wiki/Code-Style # 6